### PR TITLE
PYTHON-3297 Test auto decryption occurs after CommandSucceeded events

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: test/test_encryption.py [hunk 1]
```diff
@@ -1769,6 +1770,85 @@ def test_case_8(self):
         self.assertEqual(len(self.topology_listener.results["opened"]), 1)
 
 
+# https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests
+# /README.rst#14-decryption-events
+class TestDecryptProse(EncryptionIntegrationTest):
+    def setUp(self):
+        self.setup_client = MongoClient()
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply test/test_encryption.py]: Done.

File: test/test_encryption.py [hunk 2]
```diff
@@ -1769,6 +1770,85 @@ def test_case_8(self):
...
+            }
+        )
+        with self.assertRaises(AutoReconnect):
+            self.encrypted_client.db.decryption_events.aggregate([])
+        self.assertEqual(len(self.listener.results["failed"]), 1)
+        self.assertEqual(self.listener.results["failed"][0].command_name, "aggregate")
+
+    def test_decrypt_error(self):
+        with self.assertRaises(EncryptionError):
+            self.encrypted_client.db.decryption_events.insert_one(
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply test/test_encryption.py]: Done.

File: test/test_encryption.py [hunk 3]
```diff
@@ -1769,6 +1770,85 @@ def test_case_8(self):
...
+                {"encrypted": self.malformed_cipher_text}
+            )
+            next(self.encrypted_client.db.decryption_events.aggregate([]))
+        i = self.listener.results["succeeded"][0]
+        self.assertEqual(
+            i.reply["cursor"]["firstBatch"][0]["encrypted"], self.malformed_cipher_text
+        )
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply test/test_encryption.py:1841]: Done.

File: test/test_encryption.py [hunk 4]
```diff
@@ -1769,6 +1770,85 @@ def test_case_8(self):
...
+        self.assertEqual(len(self.listener.results["failed"]), 1)
+        self.assertEqual(self.listener.results["failed"][0].command_name, "aggregate")
+
+    def test_decrypt_error(self):
+        with self.assertRaises(EncryptionError):
+            self.encrypted_client.db.decryption_events.insert_one(
+                {"encrypted": self.malformed_cipher_text}
+            )
+            next(self.encrypted_client.db.decryption_events.aggregate([]))
+        i = self.listener.results["succeeded"][0]
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply test/test_encryption.py]: Done. here and elsewhere.

File: test/test_encryption.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply test/test_encryption.py:1841]: I don't see this added.

File: test/test_encryption.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply test/test_encryption.py]: This is not done. Please re-read my comment.

File: test/test_encryption.py [hunk 5]
```diff
@@ -1769,6 +1770,85 @@ def test_case_8(self):
...
+            key_vault_namespace="keyvault.datakeys", kms_providers=kms_providers_map
+        )
+        self.listener = AllowListEventListener("aggregate")
+        self.encrypted_client = MongoClient(
+            auto_encryption_opts=opts, retryReads=False, event_listeners=[self.listener]
+        )
+        self.addCleanup(self.encrypted_client.close)
+
+    def test_command_error(self):
+        self.setup_client.admin.command(
```
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply test/test_encryption.py]: FYI you can put multiple context managers in one with-statement. So this:
```python
with self.fail_point(...):
    with self.assertRaises(...):
        pass
```
Can be written as:
```python
with self.fail_point(...), self.assertRaises(...):
    pass
```

No need to change anything though.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Pull request merged**